src/System.Management.Automation/engine/runtime/ScriptBlockToPowerShell.cs

```diff
@@ -336,6 +336,10 @@ private ScriptBlockToPowerShellConverter()
             // Foreach-Object -Parallel call scope. This will filter the using variable map to variables 
             // only within the current (outer) Foreach-Object -Parallel call scope.
             var usingAsts = UsingExpressionAstSearcher.FindAllUsingExpressions(scriptBlock.Ast).ToList();
+            
+            // If the scriptblock ast parent is null, then it comes from a scriptblock variable and does
+            // not include the initiating Foreach-Object -Parallel command.
+            bool astIncludesForEachCommand = scriptBlock.Ast.Parent is not null;
```

Comment:
The following script also fails today, but the script block AST's parent is not null:
```
PS:9> $data = "hi"
>> $sb = { Write-Host $using:data }
>> 1 | forEach-Object -Parallel $sb
InvalidOperation: A Using variable cannot be retrieved. A Using variable can be used only with Invoke-Command, Start-Job, or InlineScript in the script workflow. When it is used with Invoke-Command, the Using variable is valid only if the script block is invoked on a remote computer.
[C:\]
PS:10> $sb.Ast.Parent

ScriptBlock                StaticType                               Extent                     Parent
-----------                ----------                               ------                     ------
{ Write-Host $using:data } System.Management.Automation.ScriptBlock { Write-Host $using:data } { Write-Host $using:data }
```
It doesn't seem this scenario is handled by this test, does it?

Reply:
Yes, it does not handle this case `$sb = { ... }`.  There may other variations as well.  I can try looking for an assignment statement as well.

Reply:
Yep, there will be other variants, such as a script block returned from a function ...
I will look closely later today and see if I can provide any suggestions.

Reply:
I am working on a fix now.  Thanks for pointing this out.